ksql-engine/src/test/java/io/confluent/ksql/TopologyFileGenerator.java

```diff
@@ -0,0 +1,74 @@
+package io.confluent.ksql;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+/**
+ * This class is used to generate the topology files to ensure safe
+ * upgrades of KSQL across releases.
+ *
+ * There are some manual steps in using this class but this should be ok as
+ * we only need to create new topology files at the end of a release cycle.
+ *
+ * The steps to generate topology files:
+ *
+ * 1. Run this class BEFORE you update the pom with a new version.  This class will generate expected topology files
+ * for the version about to be released in ksql-engine/src/test/resources/VERSION_NUM_expected_topology directory.  Where
+ * VERSION_NUM is the version defined in ksql-engine/pom.xml &lt;parent&gt;&lt;version&gt; element.
+ *
+ * 2. Update the CURRENT_TOPOLOGY_CHECKS_DIR variable in the {@link QueryTranslationTest} class with the newly generated directory name
+ * so all tests run against the newly written topology files by default.
+ *
+ */
+public class TopologyFileGenerator {
+
+    private static final String BASE_DIRECTORY = "ksql-engine/src/test/resources/";
```

Comment:
at the moment all the resources for a version are held under something like `test/resource/5_0_expected_topology`. It might be nice to store all the expected topologies until a common sub dir, e.g. `test/resources/exected_topology/5.0`. Just so the root `resources` folder doesn't get inundated with versions, and would make it easier in the future if we wanted to iterate over each version.

Reply:
ack